### ISSUE TITLE
Can't figure out how to reassign joystick inputs

### ISSUE THREAD

**Issue by BoomDoomTomb** [zajo/appler#9]
Trying to get the game "Sundog" to run on my laptop, but it appears to use joystick buttons and inputs. I read your instructions that Appler uses a virtual joystick and that the commands can be remapped, but I can't figure out how to actually _access_ that option. I press F4 to get to the virtual keyboard screen, but then my only option seems to be to press a keyboard input, which then asks me to select another keyboard input to remap it to.

I can't seem to find a way to access the joystick inputs and button 0 and button 1 from my laptop (which doesn't have a numpad). Is there a way to access the ascii remapper without first pressing a keyboard input? I just want to remap the joystick and button 0 & button 1.

Thanks!

**Comment by zajo** [comment]
You can only remap the joystick with the F4 keyboard remapper, but if I remember correctly it is able to save your setup in a file. That said, the last time I tried to use this, it didn't work. For some reason, even the simple test of reading and printing e.g. PDL(0) from basic wasn't working correctly. So it may be broken.

**Comment by BoomDoomTomb** [comment]
Thanks for your response. The problem seems to be that I can't seem to access the joystick buttons to remap them. I can see that they exist when I scroll through the ascii remapper list. But in order to change them it appears that I have to press a joystick input _first_ and then press the key I want to remap it to. Which can't happen since I don't have a joystick button to press in the first place, haha.

I do see that I can save my keyboard mapped keys to a .kbd file. But I can't figure out how to edit that file.

**Comment by zajo** [comment]
To assign a joystick function (stick or button) to a key, just find the function in the ascii map -- they are in the last column, so you have to press the right arrow until you get to it, but also the UI is stupid and it won't let you go to the last column if it doesn't have anything at the current vertical position of the cursor. So, make sure the cursor is at the top line in the table, then press the right arrow until you see the column with the joystick functions.

But again, last time I tried this, I assigned the keys correctly and it didn't work via reading PDL(0).

**Comment by BoomDoomTomb** [comment]
Can you explain how to access the ASCII map? I only seem to be able to access it by pressing a key on my keyboard, which brings it up, but prompts to remap the key I just pressed in the first place. I can't figure out how to bring up the ASCII map on its own.

**Comment by zajo** [comment]
Yes that's how it works, you press the button you want to function as if you push the joystick lever left, then select the left arrow symbol (which is in the ascii map's last column) and assign it to that key.

**Comment by BoomDoomTomb** [comment]
Ah, gotcha. Yup, it's not working, haha. Changed the Joystick buttons 0 & 1 to keyboard letters Q and E, saved the .KBD file, and they aren't registering.

**Comment by zajo** [comment]
Might be a simple thing to fix, or not. :)
